### PR TITLE
Dynamic warranty link for manufacturers' support urls

### PR DESCRIPTION
**Pull request by akemidx**
A way to dynamically produce a manufacturur's warranty link has been added.

Instead of building out code per individual manufacturer which is unscalable, this method will allow any manufacturer to have a warranty link be customized for use.

SupportURLs can now be written with `{LOCALE}` and `{SERIAL}` tags in them. Using these tags, the system will automatically replace `{LOCALE}` with the localization variable selected in settings. `{SERIAL}` will be replaced with the serial number of the asset being viewed.

For example:
Screenshot 2023-04-24 at 8 30 55 PM

-strict URL validation has been removed.
-urls must have `://` in them to validate.
-Other special characters are now allowed as long as `://` is in the link.
-No existing urls or presaved links should be affected, but will need to have `://` if the manufacturer is edited in the future.
-urls are free to not use the tags
-the individual icons for manufacturers have been changed to a default "outside link" icon next to the warranty months

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Tested in local. on existing manufacturers and with newly created manufacturers.

# Checklist:

- [x] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [x] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by akemidx** [comment]
Additionally, a byproduct of this change will open up the support url field to be used for more diverse links/clickthroughs rather than strict web urls

File: app/Http/Requests/SaveManufacturerRequest.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+     * Prepare the data for validation.
+     *
+     * @return void
+     */
+    protected function prepareForValidation()
+    {
+        $this->merge([
+            'support_url' => $this->input('support_url'),
+        ]);
+        dd($this->all());
```
Comment (snipe) [inline app/Http/Requests/SaveManufacturerRequest.php]: We probably don't want to leave a `dd()` in there

File: app/Models/Manufacturer.php [hunk 2]
```diff
@@ -23,8 +23,8 @@ class Manufacturer extends SnipeModel
     protected $rules = [
         'name'   => 'required|min:2|max:255|unique:manufacturers,name,NULL,id,deleted_at,NULL',
         'url'   => 'url|nullable',
-        'support_url'   => 'url|nullable',
         'support_email'   => 'email|nullable',
+        'support_url'   => ['regex:/.+:\/\/.+/','nullable'],
```
Comment (snipe) [inline app/Models/Manufacturer.php]: As I mentioned on the dev call, we *might* be able to simplify this a bit, using the `starts_with` validation.

`'support_url' => 'nullable|starts_with:http://,https://',afp://,facetime://,file://,irc://'`

**Review by snipe — CHANGES_REQUESTED** [review]
This looks really good - just a few changes!

File: app/Models/Manufacturer.php [hunk 3]
```diff
@@ -23,8 +23,8 @@ class Manufacturer extends SnipeModel
     protected $rules = [
         'name'   => 'required|min:2|max:255|unique:manufacturers,name,NULL,id,deleted_at,NULL',
         'url'   => 'url|nullable',
-        'support_url'   => 'url|nullable',
         'support_email'   => 'email|nullable',
+        'support_url'   => ['starts_with:http://,https://,afp://,facetime://,file://,irc://','nullable'],
```
Comment (snipe) [inline app/Models/Manufacturer.php]: Do we need to use brackets here? I don't think we do - just 

` 'support_url'   => 'nullable|starts_with:http://,https://,afp://,facetime://,file://,irc://',` 

should do it, no? I'd prefer that for consistency if it works.

File: app/Http/Controllers/ManufacturersController.php [hunk 4]
```diff
@@ -3,6 +3,7 @@
 namespace App\Http\Controllers;
 
 use App\Http\Requests\ImageUploadRequest;
+use App\Http\Requests\SaveManufacturerRequest;
```
Comment (snipe) [inline app/Http/Controllers/ManufacturersController.php]: I don't think we need this anymore?

File: app/Models/Manufacturer.php (hunk 3 above)
Comment (akemidx) [reply app/Models/Manufacturer.php]: all set

File: app/Http/Controllers/ManufacturersController.php (hunk 4 above)
Comment (akemidx) [reply app/Http/Controllers/ManufacturersController.php]: all set

**Pull request merged**